CHANGELOG.md

```diff
@@ -45,6 +46,8 @@
   inside the closure or `functools.partial` passed as `solver` argument.
 * `nk.optimizer.sr.SRLazyCG` and `nk.optimizer.sr.SRLazyGMRES` have been deprecated and will soon be removed.
 * Parts of the `Lattice` API have been overhauled, with deprecations of several methods in favor of a consistent usage of `Lattice.position` for real-space location of sites and `Lattice.basis_coords` for location of sites in terms of basis vectors. `Lattice.sites` has been added, which provides a sequence of `LatticeSite` objects combining all site properties. Furthermore, `Lattice` now provides lookup of sites from their position via `id_from_position` using a hashing scheme that works across periodic boundaries. [#703](https://github.com/netket/netket/pull/703) [#715](https://github.com/netket/netket/pull/715)
+* The `Grid` API has been retired in favour of specialised `Lattice` constructors. Since Lattice does not yet support coloured edges, the `color_edges` constructor switch has been removed for the time being. [#724](https://github.com/netket/netket/pull/724)
```

Comment:
```suggestion
* The `graph.Grid` class has been removed. {ref}`graph.Grid` will now return an instance of {ref}`graph.Lattice` supporting the same API but with new functionalities related to spatial symmetries. The `color_edges` optional keyword argument has been removed without deprecation and is slated for reintroduction in a future release. [#724](https://github.com/netket/netket/pull/724)
```

@femtobit `color_edges` was present in 2.1? I think it was.
@attila-i-szabo We tend to deprecate instead of hard-removing features unless really needed. 
I understand that we cannot support all the new fancy things inside of Lattice,  but... do we still have a way to work with coloured edges now or if we merge your PR we completely loose support for those? 
We could also make it so that `graph.grid` constructs a lattice if no `color_edges` is passed, and returns an old `graph.grid`+ deprecation warning if it's given color_edges.

cc @gcarleo which I think won't be in favour of this.

Reply:
> @femtobit color_edges was present in 2.1? I think it was.

No, I added it as part of v3. There were not colored edges in v2.1.

In fact, there was no `Grid` (no `Chain` or `Square`) in v2. Back in the old days, we even had to create our non-`Hypercube` grids by hand.

Reply:
ok, that's good for me.

Reply:
I had some chat with @femtobit about this. The big problem with the `Lattice` API is that it's too strongly tied to building the edges with KDTree. Just about any cond-mat simulation library lets you build lattices by specifying the edges yourself in a format like `(site1, site2, lattice_vector_offset) -> colour`. We need to support something like this, because it covers every possible use case. Having said that, autodetecting nearest neighbours is nice, so I'd keep that as a default behaviour if edges aren't supplied.

Reply:
@PhilipVinc @gcarleo IMO there's no point in developing two completely different APIs to do the same thing in parallel, and we're better off if we provide the `color_edges` feature for `Grid` by sorting the more general issue with `Lattice` first.

Reply:
Yes, that's something for future work.

Though note that you can mostly specify the graph with bonds and colors yourself with `CustomGraph` (and since we can pass different graphs to operators on the same Hilbert space, one can construct different graphs for different operators as well). The only things that is lacking is a notion of site location, but that one can keep in an underlying lattice. Maybe we can add an example doing this to get different bond operators in `Examples` for 3.0.

Reply:
> Though note that you can mostly specify the graph with bonds and colors yourself with CustomGraph (and since we can pass different graphs to operators on the same Hilbert space, one can construct different graphs for different operators as well). The only things that is lacking is a notion of site location, but that one can keep in an underlying lattice. Maybe we can add an example doing this to get different bond operators in Examples for 3.0.

This sounds like a very bad antipattern and not something I'd like to advertise (although users will have to figure out something like this if they want to use Netket for serious work)